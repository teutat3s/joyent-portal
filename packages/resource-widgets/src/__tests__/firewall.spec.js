/* eslint-disable camelcase */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import {
  Rules,
  TagRules,
  DefaultRules,
  ToggleFirewallForm,
  ToggleInactiveForm
} from '../firewall';

import Theme from '@mocks/theme';

const rules = [
  {
    id: '3473327e-6d0c-4747-97c5-ced1260a7e42',
    enabled: true,
    rule_str: 'FROM any TO all vms ALLOW icmp TYPE 8 CODE 0',
    rule_obj: {
      from: [['wildcard', 'any']],
      to: [['wildcard', 'vmall']],
      action: 'allow',
      protocol: {
        name: 'icmp',
        targets: ['8:0']
      },
      isWildcard: true,
      tags: []
    },
    global: true,
    description: null
  },
  {
    id: '24d55d2f-12ba-4935-9de9-ac305b8a40a5',
    enabled: true,
    rule_str: 'FROM any TO all vms ALLOW icmp6 TYPE all',
    rule_obj: {
      from: [['wildcard', 'any']],
      to: [['wildcard', 'vmall']],
      action: 'allow',
      protocol: {
        name: 'icmp6',
        targets: ['all']
      },
      isWildcard: true,
      tags: []
    },
    global: true,
    description: 'allow all ICMPv6 types'
  },
  {
    id: '4bd8b2e2-981b-474b-9b8b-0b53fecb4b71',
    enabled: false,
    rule_str: 'FROM all vms TO all vms ALLOW tcp PORT all',
    rule_obj: {
      from: [['wildcard', 'vmall']],
      to: [['wildcard', 'vmall']],
      action: 'allow',
      protocol: {
        name: 'tcp',
        targets: ['all']
      },
      isWildcard: true,
      tags: []
    },
    global: null,
    description: null
  },
  {
    id: 'af549024-b3b1-43bf-8a66-49c2b2dc5640',
    enabled: false,
    rule_str: 'FROM all vms TO all vms ALLOW tcp PORT all',
    rule_obj: {
      from: [['wildcard', 'vmall']],
      to: [['wildcard', 'vmall']],
      action: 'allow',
      protocol: {
        name: 'tcp',
        targets: ['all']
      },
      isWildcard: true,
      tags: []
    },
    global: null,
    description: null
  },
  {
    id: '9e5a152a-582b-4525-909e-f9c55deb7f03',
    enabled: false,
    rule_str: 'FROM any TO tag "wat" ALLOW tcp PORT all',
    rule_obj: {
      from: [['wildcard', 'any']],
      to: [['tag', 'wat']],
      action: 'allow',
      protocol: {
        name: 'tcp',
        targets: ['all']
      },
      isWildcard: false,
      tags: ['wat']
    },
    global: null,
    description: null
  }
];

it('renders <Rules/> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <Rules />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <Rules rules /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <Rules rules={rules} />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <DefaultRules /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <DefaultRules />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <DefaultRules rules /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <DefaultRules rules={rules} />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <TagRules /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <TagRules />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <TagRules rules /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <TagRules rules={rules} />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <ToggleFirewallForm /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <ToggleFirewallForm />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <ToggleFirewallForm submitting /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <ToggleFirewallForm submitting />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <ToggleInactiveForm /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <ToggleInactiveForm />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});
