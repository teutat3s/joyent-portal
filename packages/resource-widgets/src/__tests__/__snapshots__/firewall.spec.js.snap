// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <DefaultRules /> without throwing 1`] = `
Array [
  .c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 400;
  line-height: 1.625rem;
  font-size: 1.3125rem;
}

<h3
    className="c0"
>
    Default firewall rules
</h3>,
  .c0 {
  margin-top: 1.125rem;
}

<div
    className="c0"
/>,
]
`;

exports[`renders <DefaultRules rules /> without throwing 1`] = `
Array [
  .c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 400;
  line-height: 1.625rem;
  font-size: 1.3125rem;
}

<h3
    className="c0"
>
    Default firewall rules
</h3>,
  .c1 {
  margin-bottom: 0.75rem;
}

.c4 {
  margin-bottom: 0rem;
}

.c0 {
  margin-top: 1.125rem;
}

.c2 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
}

.c3 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
  border-color: rgb(216,216,216);
  background-color: rgb(250,250,250);
  color: rgb(73,73,73);
  cursor: default;
}

<div
    className="c0"
>
    <div
        className="c1"
    >
        <div
            className="c2"
            disabled={false}
            name="card"
        />
    </div>
    <div
        className="c1"
    >
        <div
            className="c2"
            disabled={false}
            name="card"
        />
    </div>
    <div
        className="c1"
    >
        <div
            className="c3"
            disabled={true}
            name="card"
        />
    </div>
    <div
        className="c1"
    >
        <div
            className="c3"
            disabled={true}
            name="card"
        />
    </div>
    <div
        className="c4"
    >
        <div
            className="c3"
            disabled={true}
            name="card"
        />
    </div>
</div>,
]
`;

exports[`renders <Rules rules /> without throwing 1`] = `
Array [
  .c0 {
  margin-bottom: 0.75rem;
}

.c1 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
}

<div
    className="c0"
>
    <div
        className="c1"
        disabled={false}
        name="card"
    />
</div>,
  .c0 {
  margin-bottom: 0.75rem;
}

.c1 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
}

<div
    className="c0"
>
    <div
        className="c1"
        disabled={false}
        name="card"
    />
</div>,
  .c0 {
  margin-bottom: 0.75rem;
}

.c1 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
  border-color: rgb(216,216,216);
  background-color: rgb(250,250,250);
  color: rgb(73,73,73);
  cursor: default;
}

<div
    className="c0"
>
    <div
        className="c1"
        disabled={true}
        name="card"
    />
</div>,
  .c0 {
  margin-bottom: 0.75rem;
}

.c1 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
  border-color: rgb(216,216,216);
  background-color: rgb(250,250,250);
  color: rgb(73,73,73);
  cursor: default;
}

<div
    className="c0"
>
    <div
        className="c1"
        disabled={true}
        name="card"
    />
</div>,
  .c0 {
  margin-bottom: 0rem;
}

.c1 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
  border-color: rgb(216,216,216);
  background-color: rgb(250,250,250);
  color: rgb(73,73,73);
  cursor: default;
}

<div
    className="c0"
>
    <div
        className="c1"
        disabled={true}
        name="card"
    />
</div>,
]
`;

exports[`renders <Rules/> without throwing 1`] = `null`;

exports[`renders <TagRules /> without throwing 1`] = `
Array [
  .c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 400;
  line-height: 1.625rem;
  font-size: 1.3125rem;
}

<h3
    className="c0"
>
    Firewall rules from instance tags
</h3>,
  .c0 {
  margin-top: 1.125rem;
}

<div
    className="c0"
/>,
]
`;

exports[`renders <TagRules rules /> without throwing 1`] = `
Array [
  .c0 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 400;
  line-height: 1.625rem;
  font-size: 1.3125rem;
}

<h3
    className="c0"
>
    Firewall rules from instance tags
</h3>,
  .c1 {
  margin-bottom: 0.75rem;
}

.c4 {
  margin-bottom: 0rem;
}

.c0 {
  margin-top: 1.125rem;
}

.c2 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
}

.c3 {
  box-sizing: content-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  border-width: 0.0625rem;
  border-style: solid;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  color: rgb(73,73,73);
  background-color: rgb(255,255,255);
  border-color: rgb(216,216,216);
  border-color: rgb(216,216,216);
  background-color: rgb(250,250,250);
  color: rgb(73,73,73);
  cursor: default;
}

<div
    className="c0"
>
    <div
        className="c1"
    >
        <div
            className="c2"
            disabled={false}
            name="card"
        />
    </div>
    <div
        className="c1"
    >
        <div
            className="c2"
            disabled={false}
            name="card"
        />
    </div>
    <div
        className="c1"
    >
        <div
            className="c3"
            disabled={true}
            name="card"
        />
    </div>
    <div
        className="c1"
    >
        <div
            className="c3"
            disabled={true}
            name="card"
        />
    </div>
    <div
        className="c4"
    >
        <div
            className="c3"
            disabled={true}
            name="card"
        />
    </div>
</div>,
]
`;

exports[`renders <ToggleFirewallForm /> without throwing 1`] = `
.c9 {
  margin-right: 0.75rem;
  margin-left: 0.75rem;
}

.c4 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
}

.c4[type='checkbox'],
.c4[type='radio'] {
  box-sizing: border-box;
  padding: 0;
}

.c4[type='number']::-webkit-inner-spin-button,
.c4[type='number']::-webkit-outer-spin-button {
  height: auto;
}

.c4[type='search'] {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
  outline-offset: -0.125rem;
}

.c4[type='search']::-webkit-search-cancel-button,
.c4[type='search']::-webkit-search-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4::-webkit-file-upload-button {
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  font: inherit;
}

.c8 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-style: normal;
  font-stretch: normal;
  display: block;
  color: rgb(70,70,70);
  text-align: left;
}

.c3 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  display: none;
}

.c3:checked + label::after {
  opacity: 1;
}

.c3:selected + label::after {
  opacity: 1;
}

.c3:checked + label {
  border-color: rgb(59,70,204);
}

.c3:selected + label {
  border-color: rgb(59,70,204);
}

.c3:disabled + label {
  background-color: rgb(249,249,249);
  cursor: not-allowed;
}

.c5 {
  color: rgb(100,100,100);
  position: absolute;
  width: 1.125rem;
  height: 1.125rem;
  box-sizing: border-box;
  top: 0;
  cursor: pointer;
  background-color: rgb(255,255,255);
  box-shadow: none;
  border: 0.0625rem solid rgb(216,216,216);
  cursor: pointer;
  border-radius: 0.25rem;
  width: 1.125rem;
  height: 1.125rem;
}

.c5::after {
  opacity: 0;
  content: '';
  position: absolute;
  width: 0.375rem;
  height: 0.125rem;
  background: transparent;
  top: 0.3125rem;
  left: 0.25rem;
  border: 0.125rem solid rgb(45,45,45);
  border-top: none;
  border-right: none;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c2 {
  display: inline-block;
  vertical-align: text-bottom;
  width: 1.125rem;
  height: 1.125rem;
  position: relative;
  cursor: pointer;
}

.c6 label {
  font-weight: normal;
  font-size: 0.9375rem;
}

.c1 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 1.5rem;
}

.c1 label {
  font-weight: 400;
}

.c1 label::after {
  width: 0.475rem;
  height: 0.205rem;
  top: 0.4125rem;
  left: 0.35rem;
}

.c1 div[type='checkbox'],
.c1 div[type='checkbox'] input,
.c1 div[type='checkbox'] label {
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  display: inline-block;
  padding: 0;
  border: none;
  overflow: hidden;
  height: auto;
  -webkit-padding-before: 0;
  -webkit-padding-start: 0;
  -webkit-padding-end: 0;
  -webkit-padding-after: 0;
}

.c7 {
  font-weight: 600;
  white-space: pre;
  font-size: 0.8125rem;
  cursor: pointer;
}

<form
  onChange={undefined}
>
  <div
    className="c0"
    id="fw-toggle-firewall"
    name="enabled"
    role="group"
    style={undefined}
    type="checkbox"
  >
    <li
      className="c1"
    >
      <div
        checked={false}
        className="c2"
        disabled={false}
        id="fw-toggle-firewall"
        onBlur={undefined}
        type="checkbox"
      >
        <input
          checked={false}
          className="c3 c4"
          disabled={false}
          id="l"
          onBlur={undefined}
          type="checkbox"
        />
        <label
          className="c5"
          htmlFor="l"
        />
      </div>
      <div
        checked={false}
        className="c6"
        disabled={false}
        id="fw-toggle-firewall"
        onBlur={undefined}
      >
        <label
          className="c7 c8"
          htmlFor="l"
        >
          <div
            className="c9"
          >
            Enable Firewall
          </div>
        </label>
      </div>
    </li>
  </div>
</form>
`;

exports[`renders <ToggleFirewallForm submitting /> without throwing 1`] = `
.c9 {
  margin-right: 0.75rem;
  margin-left: 0.75rem;
}

.c4 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
}

.c4[type='checkbox'],
.c4[type='radio'] {
  box-sizing: border-box;
  padding: 0;
}

.c4[type='number']::-webkit-inner-spin-button,
.c4[type='number']::-webkit-outer-spin-button {
  height: auto;
}

.c4[type='search'] {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
  outline-offset: -0.125rem;
}

.c4[type='search']::-webkit-search-cancel-button,
.c4[type='search']::-webkit-search-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4::-webkit-file-upload-button {
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  font: inherit;
}

.c8 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-style: normal;
  font-stretch: normal;
  display: block;
  color: rgb(70,70,70);
  text-align: left;
}

.c3 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  display: none;
}

.c3:checked + label::after {
  opacity: 1;
}

.c3:selected + label::after {
  opacity: 1;
}

.c3:checked + label {
  border-color: rgb(59,70,204);
}

.c3:selected + label {
  border-color: rgb(59,70,204);
}

.c3:disabled + label {
  background-color: rgb(249,249,249);
  cursor: not-allowed;
}

.c5 {
  color: rgb(100,100,100);
  position: absolute;
  width: 1.125rem;
  height: 1.125rem;
  box-sizing: border-box;
  top: 0;
  cursor: pointer;
  background-color: rgb(255,255,255);
  box-shadow: none;
  border: 0.0625rem solid rgb(216,216,216);
  cursor: pointer;
  border-radius: 0.25rem;
  width: 1.125rem;
  height: 1.125rem;
}

.c5::after {
  opacity: 0;
  content: '';
  position: absolute;
  width: 0.375rem;
  height: 0.125rem;
  background: transparent;
  top: 0.3125rem;
  left: 0.25rem;
  border: 0.125rem solid rgb(45,45,45);
  border-top: none;
  border-right: none;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c2 {
  display: inline-block;
  vertical-align: text-bottom;
  width: 1.125rem;
  height: 1.125rem;
  position: relative;
  cursor: pointer;
}

.c6 label {
  font-weight: normal;
  font-size: 0.9375rem;
}

.c1 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 1.5rem;
}

.c1 label {
  font-weight: 400;
}

.c1 label::after {
  width: 0.475rem;
  height: 0.205rem;
  top: 0.4125rem;
  left: 0.35rem;
}

.c1 div[type='checkbox'],
.c1 div[type='checkbox'] input,
.c1 div[type='checkbox'] label {
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  display: inline-block;
  padding: 0;
  border: none;
  overflow: hidden;
  height: auto;
  -webkit-padding-before: 0;
  -webkit-padding-start: 0;
  -webkit-padding-end: 0;
  -webkit-padding-after: 0;
}

.c7 {
  font-weight: 600;
  white-space: pre;
  font-size: 0.8125rem;
  cursor: pointer;
}

<form
  onChange={undefined}
>
  <div
    className="c0"
    id="fw-toggle-firewall"
    name="enabled"
    role="group"
    style={undefined}
    type="checkbox"
  >
    <li
      className="c1"
    >
      <div
        checked={false}
        className="c2"
        disabled={true}
        id="fw-toggle-firewall"
        onBlur={undefined}
        type="checkbox"
      >
        <input
          checked={false}
          className="c3 c4"
          disabled={true}
          id="n"
          onBlur={undefined}
          type="checkbox"
        />
        <label
          className="c5"
          htmlFor="n"
        />
      </div>
      <div
        checked={false}
        className="c6"
        disabled={true}
        id="fw-toggle-firewall"
        onBlur={undefined}
      >
        <label
          className="c7 c8"
          htmlFor="n"
        >
          <div
            className="c9"
          >
            Enable Firewall
          </div>
        </label>
      </div>
    </li>
  </div>
</form>
`;

exports[`renders <ToggleInactiveForm /> without throwing 1`] = `
.c9 {
  margin-right: 0.75rem;
  margin-left: 0.75rem;
}

.c4 {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
}

.c4[type='checkbox'],
.c4[type='radio'] {
  box-sizing: border-box;
  padding: 0;
}

.c4[type='number']::-webkit-inner-spin-button,
.c4[type='number']::-webkit-outer-spin-button {
  height: auto;
}

.c4[type='search'] {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
  outline-offset: -0.125rem;
}

.c4[type='search']::-webkit-search-cancel-button,
.c4[type='search']::-webkit-search-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4::-webkit-file-upload-button {
  -webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  font: inherit;
}

.c8 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-style: normal;
  font-stretch: normal;
  display: block;
  color: rgb(70,70,70);
  text-align: left;
}

.c3 {
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
  display: none;
}

.c3:checked + label::after {
  opacity: 1;
}

.c3:selected + label::after {
  opacity: 1;
}

.c3:checked + label {
  border-color: rgb(59,70,204);
}

.c3:selected + label {
  border-color: rgb(59,70,204);
}

.c3:disabled + label {
  background-color: rgb(249,249,249);
  cursor: not-allowed;
}

.c5 {
  color: rgb(100,100,100);
  position: absolute;
  width: 1.125rem;
  height: 1.125rem;
  box-sizing: border-box;
  top: 0;
  cursor: pointer;
  background-color: rgb(255,255,255);
  box-shadow: none;
  border: 0.0625rem solid rgb(216,216,216);
  cursor: pointer;
  border-radius: 0.25rem;
  width: 1.125rem;
  height: 1.125rem;
}

.c5::after {
  opacity: 0;
  content: '';
  position: absolute;
  width: 0.375rem;
  height: 0.125rem;
  background: transparent;
  top: 0.3125rem;
  left: 0.25rem;
  border: 0.125rem solid rgb(45,45,45);
  border-top: none;
  border-right: none;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c2 {
  display: inline-block;
  vertical-align: text-bottom;
  width: 1.125rem;
  height: 1.125rem;
  position: relative;
  cursor: pointer;
}

.c6 label {
  font-weight: normal;
  font-size: 0.9375rem;
}

.c1 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 1.5rem;
}

.c1 label {
  font-weight: 400;
}

.c1 label::after {
  width: 0.475rem;
  height: 0.205rem;
  top: 0.4125rem;
  left: 0.35rem;
}

.c1 div[type='checkbox'],
.c1 div[type='checkbox'] input,
.c1 div[type='checkbox'] label {
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  display: inline-block;
  padding: 0;
  border: none;
  overflow: hidden;
  height: auto;
  -webkit-padding-before: 0;
  -webkit-padding-start: 0;
  -webkit-padding-end: 0;
  -webkit-padding-after: 0;
}

.c7 {
  font-weight: 600;
  white-space: pre;
  font-size: 0.8125rem;
  cursor: pointer;
}

<form>
  <div
    className="c0"
    id="fw-toggle-inactive"
    name="inactive"
    role="group"
    style={undefined}
    type="checkbox"
  >
    <li
      className="c1"
    >
      <div
        checked={false}
        className="c2"
        disabled={false}
        id="fw-toggle-inactive"
        onBlur={undefined}
        type="checkbox"
      >
        <input
          checked={false}
          className="c3 c4"
          disabled={false}
          id="p"
          onBlur={undefined}
          type="checkbox"
        />
        <label
          className="c5"
          htmlFor="p"
        />
      </div>
      <div
        checked={false}
        className="c6"
        disabled={false}
        id="fw-toggle-inactive"
        onBlur={undefined}
      >
        <label
          className="c7 c8"
          htmlFor="p"
        >
          <div
            className="c9"
          >
            Show inactive rules
          </div>
        </label>
      </div>
    </li>
  </div>
</form>
`;
